### PR TITLE
ci: add simple integration test

### PR DESCRIPTION
**Pull request by smoya**
**Description**
This PR adds a simple integration test useful for CI which will ensure the generator works as a whole.

- Adds a new script called `test:integration`, which basically generates a test spec (Streetlights tutorial located at https://bit.ly/asyncapi) by using the `cli.js`.
- It runs the new `test:integration` script in the `test` script so the [CI workflow](https://github.com/asyncapi/generator/blob/master/.github/workflows/if-nodejs-pr-testing.yml) runs it on each PR.

**Related issue(s)**
The new test will catch dependency issues like https://github.com/asyncapi/generator/pull/510

### CODE REVIEW

**Comment by derberg** [comment]
@smoya the error you're getting atm on windows is related to a bug for windows and nvm (GA uses nvm) and unfortunately can be fixed only with https://github.com/asyncapi/ts-nats-template/blob/master/package.json#L34 as `npmi` that we use in generator for templates installation is not actively maintained 😞

**Comment by smoya** [comment]
Thanks @derberg for pointing that out. I'm gonna add that hack into the package.json scripts then so it can work properly on Windows.

**Review by fmvilas — APPROVED** [review]
Nooice 🚫 🧊

**Comment by derberg** [comment]
@smoya for the conventional commits scopes are rather for packages, parts of the project, not for stuff like ci, so my recommendation is to do prefix `test:` or `ci:`

File: package.json [hunk 1]
```diff
@@ -12,8 +12,9 @@
     "npm": ">6.13.7"
   },
   "scripts": {
-    "test": "jest --coverage",
+    "test": "jest --coverage && npm run test:integration",
```
Comment (derberg) [inline package.json:15]: what if we introduce also `test:unit` for `jest --coverage`?

while developing I often run `npm test` but now will stop as I do not want to run integration every time and best would be to have `npm run test:unit` too. We have a similar approach in the parser https://github.com/asyncapi/parser-js/blob/master/package.json

Comment (smoya) [reply package.json:15]: Completely agree 👍  Changing!

**Comment by smoya** [comment]
> @smoya for the conventional commits scopes are rather for packages, parts of the project, not for stuff like ci, so my recommendation is to do prefix `test:` or `ci:`

@derberg Thanks for the clarification. I will made the change into the squashed commit before merging.

**Review by fmvilas — APPROVED** [review]

**Pull request merged**